### PR TITLE
Export StartSpanFromContextWithTracer

### PR DESCRIPTION
**Pull request by devoxel**
Originally discussed in  #149 . This is a fairly simple PR. It just exports a helper function that already exists.

It can be useful for people not using the global tracer. It's only a small thing really, but still, could be nice.

### CODE REVIEW

**Comment by yurishkuro** [comment]
@bhs @tedsuo any objections to exposing this method?

We could also simplify the name to just `opentracing.StartSpan(context.Context, Tracer)`

**Comment by bhs** [comment]
@yurishkuro 

> We could also simplify the name to just opentracing.StartSpan(context.Context, Tracer)

Might be a little odd / asymmetric given the presence of `opentracing.StartSpanFromContext`... I don't have strong feelings, though.

**Comment by devoxel** [comment]
The name does seem a bit long. I was wondering, is there any strong reason why Tracer isn't potentially a SpanOption? It could potentially be nice to have the same API across the board.

File: gocontext.go [hunk 1]
```diff
@@ -41,11 +41,18 @@ func SpanFromContext(ctx context.Context) Span {
...
-func startSpanFromContextWithTracer(ctx context.Context, tracer Tracer, operationName string, opts ...StartSpanOption) (Span, context.Context) {
+// StartSpanFromContextWithTracer starts and returns a span with `operationName`
+// using  a span found within the context as a ChildOfRef. If that doesn't exist
+// it creates a root span. It also returns a context.Context object built
+// around the returned span.
+//
+// It's behavior is identical to StartSpanFromContext except that it takes an explicit
+// tracer as opposed to using the global tracer.
+func StartSpanFromContextWithTracer(ctx context.Context, tracer Tracer, operationName string, opts ...StartSpanOption) (Span, context.Context) {
+	var span Span
```
Comment (yurishkuro) [inline gocontext.go]: build fails because this statement is not needed

Comment (pritishshah) [reply gocontext.go]: @devoxel would you please fix it?

**Pull request closed without merging**